sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/reflect/ByteBuddyDoFnInvokerFactory.java

```diff
@@ -554,15 +487,13 @@ public StackManipulation dispatch(RestrictionTrackerParameter p) {
             // but the @ProcessElement method expects a concrete subtype of it.
             // Insert a downcast.
             return new StackManipulation.Compound(
-                simpleExtraContextParameter("restrictionTracker", pushExtraContextFactory),
+                simpleExtraContextParameter("restrictionTracker"),
```

Comment:
I feel like the use of strings here is dangerous. Can we use constants or something (maybe an enum?)